sparkles/tests/test_review.py

```diff
@@ -182,3 +184,113 @@ def test_run_aca_review_function():
+          'date': '2019:073:21:55:30.000',
+          'detector': 'ACIS-S',
+          'dither_acq': (7.9992, 7.9992),
+          'dither_guide': (7.9992, 7.9992),
+          'focus_offset': 0.0,
+          'man_angle': 122.97035882921071,
+          'n_acq': 8,
+          'n_fid': 0,
+          'n_guide': 8,
+          'obsid': 48334.0,
+          'sim_offset': 0.0,
+          't_ccd_acq': -10.257559323423214,
+          't_ccd_guide': -10.25810835536192}
+    aca = get_aca_catalog(**kw)
+    acar = aca.get_review_table()
+    acar.get_roll_options()
+    assert Quat(kw['att']).roll <= acar.roll_info['roll_max']
+    assert Quat(kw['att']).roll >= acar.roll_info['roll_min']
+
+
+def test_calc_targ_from_aca():
+    """
+    Confirm _calc_targ_from_aca seems to do the right thing based on obsid
+    This does a bit too much processing for what should be a lightweight test.
+    """
+    obs_kwargs = KWARGS_48464
+    aca_er = get_aca_catalog(**obs_kwargs)
+    acar_er = aca_er.get_review_table()
+    q_targ = acar_er._calc_targ_from_aca(acar_er.att, 0, 0)
+    # Cheat and use the string reprs for comparison to 8 decimals
+    assert Quat(obs_kwargs['att']).__repr__() == q_targ.__repr__()
+
+    or_obs_kwargs = obs_kwargs.copy()
+    or_obs_kwargs['obsid'] = 1
+    aca_or = get_aca_catalog(**or_obs_kwargs)
+    acar_or = aca_or.get_review_table()
+    q_targ = acar_or._calc_targ_from_aca(acar_or.att, 0, 0)
+    # Assert that the the OR version *doesn't* match and is off by about 70 arcsecs yaw
+    assert Quat(obs_kwargs['att']).__repr__() != q_targ.__repr__()
+    assert abs(Quat(obs_kwargs['att']).dq(q_targ).yaw * 3600) - 69.59 < .01
+
+
+def test_get_roll_intervals():
+    """
+    Test that the behavior of get_roll_intervals is different for ORs and ERs with
+    regard to use of offsets.  They are set to arbitrary large values in the test.
+    """
+
+    # This uses the catalog at KWARGS_48464, but would really be better as a fully
+    # synthetic test
+    obs_kwargs = KWARGS_48464
+    aca_er = get_aca_catalog(**obs_kwargs)
+    acar_er = aca_er.get_review_table()
+
+    kw_or = obs_kwargs.copy()
+    # Set this one to have an OR obsid (and not 0 which is special)
+    kw_or['obsid'] = 1
+    aca_or = get_aca_catalog(**kw_or)
+    acar_or = aca_or.get_review_table()
+
+    # Use these values to override the get_roll_intervals ranges to get more interesting
+    # outputs.  y_off and z_off are really 0 everywhere for now from ORViewer though.
+    y_off = 20 / 60.
+    z_off = 30 / 60.
+    roll_dev = 5
+
+    er_roll_intervs, er_info = acar_er.get_roll_intervals(
+        acar_er.get_candidate_better_stars(),
+        roll_dev=roll_dev, y_off=y_off, z_off=z_off)
+
+    or_roll_intervs, or_info = acar_or.get_roll_intervals(
+        acar_or.get_candidate_better_stars(),
+        roll_dev=roll_dev, y_off=y_off, z_off=z_off)
+
+    assert Quat(obs_kwargs['att']).roll <= er_info['roll_max']
```

Comment:
Use `acar_er.att` which is guaranteed to be a Quat.

Reply:
I wasn't sure if that was cheating, but OK.